packages/admin/admin/src/FinalForm.tsx

```diff
@@ -173,8 +173,13 @@ export function FinalForm<FormValues = AnyObject>(props: IProps<FormValues>) {
                         if (formRenderProps.hasValidationErrors) {
                             return false;
                         }
+
+                        editDialogFormApi?.onFormStatusChange("saving");
+                        const validationErrors = await waitForFormValidationToBeFinished(formRenderProps.form);
                         const submissionErrors = await formRenderProps.form.submit();
-                        if (submissionErrors) {
+
+                        if (submissionErrors || validationErrors) {
```

Comment:
Shouldn't we check first for hasValidationErrors, and only call submit if there are no validation errors?

Reply:
Done https://github.com/vivid-planet/comet/pull/1414/commits/675e9aad5e2616d6753dc82e6034cecfa2c8f186